### PR TITLE
Add test for printing chunks with multiple heaps

### PR DESCRIPTION
**Pull request by irontigran**
This test checks that when a non-main arena has multiple (and therefore
non-contiguous) heaps, we still print them all out.

### Description/Motivation/Screenshots ###

As part of the ongoing effort to fix the `heap chunks` for non-main arenas, this PR adds a test that triggers the creation of a new heap in a non-main arena. This new heap will be non-contiguous with the original heap, which is a corner case that our eventual solution for the `heap chunks` command will have to handle.

Originally discussed at #709.

### How Has This Been Tested? ###

| Architecture |          Yes/No          | Comments                                  |
| ------------ | :----------------------: | ----------------------------------------- |
| x86-32       | :heavy_check_mark: | Replace with :heavy_check_mark: if tested |
| x86-64       | :heavy_check_mark: |                                           |
| ARM          | :heavy_multiplication_x: |                                           |
| AARCH64      | :heavy_multiplication_x: |                                           |
| MIPS         | :heavy_multiplication_x: |                                           |
| POWERPC      | :heavy_multiplication_x: |                                           |
| SPARC        | :heavy_multiplication_x: |                                           |
| RISC-V       | :heavy_multiplication_x: |                                           |
| `make test`  | :heavy_check_mark: |                                           |

### Checklist ###

- [x] My PR was done against the `dev` branch, not `master`.
- [x] My code follows the code style of this project.
- [x] My change includes a change to the documentation, if required.
- [x] My change adds tests as appropriate.
- [x] I have read and agree to the **CONTRIBUTING** document.

### CODE REVIEW

File: tests/binaries/heap-multiple-heaps.c
Comment (Grazfather) [inline tests/binaries/heap-multiple-heaps.c:13]: Where does this number come from?

**Review by Grazfather — COMMENTED** [review]
Awesome. Much appreciated.

File: tests/binaries/heap-multiple-heaps.c
Comment (irontigran) [reply tests/binaries/heap-multiple-heaps.c:13]: It's something of an appendix from initial experiments. When I created two threads (and therefore two arenas), I noticed that the second arena was 0x8000000 bytes away from the first arena. 0x8000000 / 127KB = ~1032, so I used that as a starting point for the number of allocations to create. As it turns out, that was bigger than needed - it only took 515 allocations to trigger the creation of a new heap, so I just left it as a "twice as big as the number of allocations I expect" limit.

I can either rename it to make its purpose (and arbitrary nature) clear, or rewrite the loop as a while loop like this:

```c
void *current, *previous;
current = previous = NULL;
while (1) {
    current = malloc(LESS_THAN_MMAP_THRESHOLD);
    int chunk_distance = (previous == NULL) ? current - previous : EXPECTED_CHUNK_DISTANCE;
    if (chunk_distance < 0 || chunk_distance > EXPECTED_CHUNK_DISTANCE) {
        __builtin_trap();
        break;
    }
}
```

Comment (Grazfather) [reply tests/binaries/heap-multiple-heaps.c:13]: name is fine, value is fine, maybe you can clarify why ~500 allocations makes you choose 1032. e.g.

```c
/* Experimentally, a new heap is created at ~500 allocations. We double for a safe margin */
```

**Comment by Grazfather** [comment]
Looks good! CI fails as you'd expect. After we merge #709 let's get you to rebase, ensure it passes CI, and then merge.

**Review by hugsy — APPROVED** [review]

**Comment by hugsy** [comment]
@irontigran You should be ok to rebase with `dev`. Once the CI passes, it can be squashed and merged

**Comment by irontigran** [comment]
The CI shouldn't pass until we've updated the `heap chunks` algorithm. Are we holding off on merging until we update `heap chunks` to handle multiple heaps?

**Comment by theguy147** [comment]
PR #716 should solve the multiple heap per arena issue and this test works fine with it after fixing the issue that when the `before` cmds are being executed the heap hasn't been initialized yet (also I renamed `get_main_arena()` to `get_glibc_arena()`.)

I made these changes locally to get it to work:
```diff
diff --git a/tests/runtests.py b/tests/runtests.py
index 9ab3080..ec69304 100755
--- a/tests/runtests.py
+++ b/tests/runtests.py
@@ -225,10 +225,10 @@ class TestGefCommandsUnit(GefUnitTestGeneric):
         return
 
     def test_cmd_heap_chunks_mult_heaps(self):
-        before = ['python gdb.execute("heap set-arena {}".format(get_main_arena().next))']
+        before = ['run', 'python gdb.execute("heap set-arena {}".format(get_glibc_arena().next))']
         cmd = "heap chunks"
         target = "/tmp/heap-multiple-heaps.out"
-        res = gdb_run_silent_cmd(cmd, before=before, target=target)
+        res = gdb_run_cmd(cmd, before=before, target=target)
         self.assertNoException(res)
         self.assertIn("Chunk(addr=", res)
         self.assertIn("top chunk", res)
```

**Comment by theguy147** [comment]
@irontigran can you rebase the PR. It should work now since #716 was merged.

**Review by theguy147 — APPROVED** [review]

File: tests/binaries/heap-multiple-heaps.c
Comment (theguy147) [reply tests/binaries/heap-multiple-heaps.c:19]: @Grazfather can this be resolved?

**Pull request merged**
